source/common/http/http1/conn_pool.cc

```diff
@@ -209,25 +215,44 @@ void ConnPoolImpl::onResponseComplete(ActiveClient& client) {
     host_->cluster().stats().upstream_cx_max_requests_.inc();
     onDownstreamReset(client);
   } else {
-    processIdleClient(client);
+    processIdleClient(client, true);
```

Comment:
Can we comment why, perhaps with a link to the original issue to make it clear why we are doing this?